// @shopify/cli-kit/dist/private/node/analytics.js
import { hashString } from '../../public/node/crypto.js';
import { getPackageManager, packageManagerUsedForCreating } from '../../public/node/node-package-manager.js';
import * as metadata from '../../metadata.js';
import { macAddress } from '../../environment/local.js';
import { platformAndArch } from '../../public/node/os.js';
import * as environment from '../../environment.js';
export async function startAnalytics({ commandContent, args, currentTime = new Date().getTime(), commandClass, }) {
    let startCommand = commandContent.command;
    if (commandClass && Object.prototype.hasOwnProperty.call(commandClass, 'analyticsNameOverride')) {
        startCommand = commandClass.analyticsNameOverride() ?? commandContent.command;
    }
    await metadata.addSensitive(() => ({
        commandStartOptions: {
            startTime: currentTime,
            startCommand,
            startArgs: args,
        },
    }));
    await metadata.addPublic(() => ({
        cmd_all_launcher: packageManagerUsedForCreating(),
        cmd_all_alias_used: commandContent.alias,
        cmd_all_topic: commandContent.topic,
        cmd_all_plugin: commandClass?.plugin?.name,
    }));
}
export async function getEnvironmentData(config) {
    const ciPlatform = environment.local.ciPlatform();
    const pluginNames = getPluginNames(config);
    const shopifyPlugins = pluginNames.filter((plugin) => plugin.startsWith('@shopify/'));
    const { platform, arch } = platformAndArch();
    return {
        uname: `${platform} ${arch}`,
        env_ci: ciPlatform.isCI,
        env_ci_platform: ciPlatform.name,
        env_plugin_installed_any_custom: pluginNames.length !== shopifyPlugins.length,
        env_plugin_installed_shopify: JSON.stringify(shopifyPlugins),
        env_shell: config.shell,
        env_web_ide: environment.local.cloudEnvironment().editor
            ? environment.local.cloudEnvironment().platform
            : undefined,
        env_device_id: hashString(await macAddress()),
        env_cloud: environment.local.cloudEnvironment().platform,
        env_package_manager: await getPackageManager(process.cwd()),
    };
}
export async function getSensitiveEnvironmentData(config) {
    return {
        env_plugin_installed_all: JSON.stringify(getPluginNames(config)),
    };
}
function getPluginNames(config) {
    return config.plugins
        .map((plugin) => plugin.name)
        .sort()
        .filter((plugin) => !plugin.startsWith('@oclif/'));
}
