// @netlify/hydrogen-platform/dist/plugin.mjs
var __require = /* @__PURE__ */ ((x) => typeof require !== "undefined" ? require : typeof Proxy !== "undefined" ? new Proxy(x, {
  get: (a, b) => (typeof require !== "undefined" ? require : a)[b]
}) : x)(function(x) {
  if (typeof require !== "undefined")
    return require.apply(this, arguments);
  throw new Error('Dynamic require of "' + x + '" is not supported');
});

// src/plugin.ts
import netlifyPlugin from "@netlify/vite-plugin-netlify-edge";
var plugin = () => {
  let resolvedConfig;
  const platformEntrypoint = __require.resolve(
    "@netlify/hydrogen-platform/handler"
  );
  return [
    netlifyPlugin(),
    {
      name: "vite-plugin-netlify-hydrogen",
      config(config) {
        if (config?.build?.ssr) {
          return {
            build: {
              ssr: platformEntrypoint,
              outDir: ".netlify/edge-functions/handler"
            }
          };
        }
      },
      configResolved(config) {
        resolvedConfig = config;
      }
    }
  ];
};
var plugin_default = plugin;
export {
  plugin_default as default
};
