// @netlify/hydrogen-platform/dist/plugin.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/plugin.ts
var plugin_exports = {};
__export(plugin_exports, {
  default: () => plugin_default
});
module.exports = __toCommonJS(plugin_exports);
var import_vite_plugin_netlify_edge = __toESM(require("@netlify/vite-plugin-netlify-edge"));
var plugin = () => {
  let resolvedConfig;
  const platformEntrypoint = require.resolve("@netlify/hydrogen-platform/handler");
  return [
    (0, import_vite_plugin_netlify_edge.default)(),
    {
      name: "vite-plugin-netlify-hydrogen",
      config(config) {
        if (config?.build?.ssr) {
          return {
            build: {
              ssr: platformEntrypoint,
              outDir: ".netlify/edge-functions/handler"
            }
          };
        }
      },
      configResolved(config) {
        resolvedConfig = config;
      }
    }
  ];
};
var plugin_default = plugin;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {});
